File: app/models/game.py
class Game:
    def __init__(self):
        self.board = [['' for _ in range(3)] for _ in range(3)]
        self.current_player = 'X'
        self.winner = None
        self.game_over = False
        self.moves = []

    def make_move(self, row, col):
        if self.game_over or self.board[row][col] != '':
            return False
        
        self.board[row][col] = self.current_player
        self.moves.append((row, col, self.current_player))
        
        if self.check_winner():
            self.winner = self.current_player
            self.game_over = True
        elif self.check_draw():
            self.game_over = True
        else:
            self.current_player = 'O' if self.current_player == 'X' else 'X'
        
        return True

    def check_winner(self):
        # Check rows
        for row in self.board:
            if row[0] != '' and row[0] == row[1] == row[2]:
                return True

        # Check columns
        for col in range(3):
            if self.board[0][col] != '' and self.board[0][col] == self.board[1][col] == self.board[2][col]:
                return True

        # Check diagonals
        if self.board[0][0] != '' and self.board[0][0] == self.board[1][1] == self.board[2][2]:
            return True
        if self.board[0][2] != '' and self.board[0][2] == self.board[1][1] == self.board[2][0]:
            return True

        return False

    def check_draw(self):
        return all(cell != '' for row in self.board for cell in row)

    def reset(self):
        self.board = [['' for _ in range(3)] for _ in range(3)]
        self.current_player = 'X'
        self.winner = None
        self.game_over = False
        self.moves = [] 

File: app/static/js/game.js
document.addEventListener('DOMContentLoaded', () => {
    const cells = document.querySelectorAll('.cell');
    const status = document.getElementById('status');
    const resetButton = document.getElementById('reset');
    const playerScores = {
        'X': document.querySelector('.player-score:nth-child(1) .score'),
        'O': document.querySelector('.player-score:nth-child(2) .score')
    };
    let gameOver = false;
    let scores = { 'X': 0, 'O': 0 };

    cells.forEach(cell => {
        cell.addEventListener('click', handleCellClick);
    });

    resetButton.addEventListener('click', resetGame);

    function handleCellClick(e) {
        if (gameOver) return;

        const cell = e.target.closest('.cell');
        const row = cell.dataset.row;
        const col = cell.dataset.col;

        fetch('/move', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({ row, col })
        })
        .then(response => response.json())
        .then(data => {
            if (data.success) {
                updateBoard(data);
                if (data.game_over && data.winner) {
                    updateScores(data.winner);
                }
            }
        });
    }

    function updateBoard(data) {
        const board = data.board;
        cells.forEach(cell => {
            const row = cell.dataset.row;
            const col = cell.dataset.col;
            const value = board[row][col];
            const cellContent = cell.querySelector('.cell-content');
            
            cellContent.textContent = '';
            cell.classList.remove('x', 'o');
            
            if (value) {
                cell.classList.add(value.toLowerCase());
            }
        });

        if (data.game_over) {
            gameOver = true;
            if (data.winner) {
                status.innerHTML = `<i class="fas fa-trophy"></i> Player ${data.winner} wins!`;
                status.parentElement.classList.remove('alert-info');
                status.parentElement.classList.add('alert-success');
            } else {
                status.innerHTML = `<i class="fas fa-handshake"></i> It's a draw!`;
                status.parentElement.classList.remove('alert-info');
                status.parentElement.classList.add('alert-warning');
            }
        } else {
            status.innerHTML = `<i class="fas fa-user"></i> Player ${data.current_player}'s turn`;
            status.parentElement.classList.remove('alert-success', 'alert-warning');
            status.parentElement.classList.add('alert-info');
        }
    }

    function updateScores(winner) {
        scores[winner]++;
        playerScores[winner].textContent = scores[winner];
        playerScores[winner].classList.add('score-updated');
        setTimeout(() => {
            playerScores[winner].classList.remove('score-updated');
        }, 1000);
    }

    function resetGame() {
        fetch('/reset', {
            method: 'POST'
        })
        .then(response => response.json())
        .then(data => {
            if (data.success) {
                cells.forEach(cell => {
                    const cellContent = cell.querySelector('.cell-content');
                    cellContent.textContent = '';
                    cell.classList.remove('x', 'o');
                });
                status.innerHTML = `<i class="fas fa-user"></i> Player X's turn`;
                status.parentElement.classList.remove('alert-success', 'alert-warning');
                status.parentElement.classList.add('alert-info');
                gameOver = false;
            }
        });
    }
}); 